mysql: switch from MySQL to Mariadb

micmac1 opened the pull request:
Maintainer: @jow- 
Compile tested: i386 target
Run tested: N/A

Description:
Hello @jow-,

My apologies in advance for this "surprise" pull request. I realized a few days ago that the mysql version in the tree is quite dated and looked at updating it. In the end I ended up replacing mysql with mariadb. But because a long time has passed between the older mysql release and now, a lot of things changed. One of them is that now cmake is used to build the source. Here some more notes I took earlier:

- Remove mysql and replace it with mariadb. Other distros have shifted their
  preference to mariadb as well, like Debian.

- mariadb is a drop-in replacement. Other software should not see any
  difference. I've compile-tested some of them (e.g. telephony packages,
  strongswan and gnunet) and there was no issue. Well, almost, see below.

- I copied *a lot* from Buildroot and Alpine Linux :-)

- autoconf is not available anymore, now cmake is used, which is nice.

- Compiling against uclibc++ is unfortunately not possible anymore, so the
  libstdcpp dep had to be added. I didn't test but my best guess is that
  current mysql versions won't compile against uclibc++ either. Luckily the
  dep is not needed for the client library, only for the server package.

- The server package is way bigger. Previously the binaries in it were linked
  against libmysqlclient, but not anymore. They all carry a full static lib
  and there's nothing to be done about it. See these links from 2011/2012 for
  explanation:

  https://bugs.launchpad.net/ubuntu/+source/mysql-5.5/+bug/898340
  https://bugs.mysql.com/bug.php?id=64570

  The server ipk goes up from 2 MB to 8 MB if I remember correctly.

- Maybe one should label the server package BROKEN on anything other than
  maybe i386 and x86_64 to avoid building the server on the buildbots. Who's
  going to install it on a tiny router? Most people probably only want
  the library anyway.

- Host tools need to be build before being able to compile mariadb for the
  target arch. I didn't install the tools into the host staging area and just
  refer to host pkg build dir in the Makefile. This works fine at home, but
  I'm not sure that this works on the build bots. If not - some sed script and
  copying a handful of tools to the host staging area should to it.

- I added the mysqld plugins in case anyone wants them.

- I honestly did not run-test this one bit. I never used mysql in my life. The
  server package I couldn't install on my router even if I wanted to, because
  it's too big. :-) I just felt updating mysql is a good exercise because the
  available version is waaay old.

- There is not longer a need for a threadsafe version of the client lib. The
  default lib _is_ threadsafe. The Makefile uses PROVIDES so that other
  packages that depend on libmysqlclient-r do not need to be updated.

- During compile tests I found that python-mysql doesn't build anymore. But
  it looks like the project is dead upstream, no commits since 2014. Don't see
  why that would hold us back. python-mysql is a depend of seafile-server,
  though.

I don't expect that this gets merged quickly or at all. I just thought I'll raise the pull request and then we'll see what comes out of it.

Kind regards,
Seb

micmac1 commented:
Hello @jow- 

I fiddled with this some more and found some compile issues with musl and also some issues with mips. But Debian and Alpine Linux already had patches for these. And in the end I tried the server out on my mips router.

```
root@hank2:/tmp# ./mysql_install_db --basedir=/usr --force
Installing MariaDB/MySQL system tables in '/tmp/mysql/data' ...
2018-04-02 22:36:00 1998318736 [Note] Using unique option prefix 'key_buffer' is error-prone and can break in the future. Please use the full name 'key_buffer_size' instead.
2018-04-02 22:36:00 1998318736 [Note] /usr/bin/mysqld (mysqld 10.1.32-MariaDB) starting as process 2420 ...
2018-04-02 22:36:00 1998318736 [ERROR] mysqld: Out of memory (Needed 130760704 bytes)
2018-04-02 22:36:00 1998318736 [ERROR] mysqld: Out of memory (Needed 98058240 bytes)
2018-04-02 22:36:00 1998318736 [ERROR] mysqld: Out of memory (Needed 73531392 bytes)
2018-04-02 22:36:00 1998318736 [Note] InnoDB: innodb_empty_free_list_algorithm has been changed to legacy because of small buffer pool size. In order to use backoff, increase buffer pool at least up to 20MB.

2018-04-02 22:36:00 1998318736 [Note] InnoDB: Using mutexes to ref count buffer pool pages
2018-04-02 22:36:00 1998318736 [Note] InnoDB: The InnoDB memory heap is disabled
2018-04-02 22:36:00 1998318736 [Note] InnoDB: Mutexes and rw_locks use GCC atomic builtins
2018-04-02 22:36:00 1998318736 [Note] InnoDB: GCC builtin __atomic_thread_fence() is used for memory barrier
2018-04-02 22:36:00 1998318736 [Note] InnoDB: Compressed tables use zlib 1.2.11
2018-04-02 22:36:00 1998318736 [Note] InnoDB: Using Linux native AIO
2018-04-02 22:36:00 1998318736 [Note] InnoDB: Using generic crc32 instructions
2018-04-02 22:36:00 771bec90 InnoDB: Error: Linux Native AIO interface is not supported on this platform. Please check your OS documentation and install appropriate binary of InnoDB.
InnoDB: You can disable Linux Native AIO by setting innodb_use_native_aio = 0 in my.cnf
2018-04-02 22:36:00 1998318736 [Warning] InnoDB: Linux Native AIO disabled.
2018-04-02 22:36:00 1998318736 [Note] InnoDB: Initializing buffer pool, size = 128.0M
InnoDB: mmap(138543104 bytes) failed; errno 12
2018-04-02 22:36:00 1998318736 [ERROR] InnoDB: Cannot allocate memory for the buffer pool
2018-04-02 22:36:00 1998318736 [ERROR] Plugin 'InnoDB' init function returned error.
2018-04-02 22:36:00 1998318736 [ERROR] Plugin 'InnoDB' registration as a STORAGE ENGINE failed.
2018-04-02 22:36:00 1998318736 [ERROR] Unknown/unsupported storage engine: InnoDB
2018-04-02 22:36:00 1998318736 [ERROR] Aborting


Installation of system tables failed!  Examine the logs in
/tmp/mysql/data for more information.

The problem could be conflicting information in an external
my.cnf files. You can ignore these by doing:

    shell> ./mysql_install_db --defaults-file=~/.my.cnf

You can also try to start the mysqld daemon with:

    shell> /usr/bin/mysqld --skip-grant-tables --general-log &

and use the command line tool /usr/bin/mysql
to connect to the mysql database and look at the grant tables:

    shell> /usr/bin/mysql -u root mysql
    mysql> show tables;

Try 'mysqld --help' if you have problems with paths.  Using
--general-log gives you a log in /tmp/mysql/data that may be helpful.

The latest information about mysql_install_db is available at
https://mariadb.com/kb/en/installing-system-tables-mysql_install_db
You can find the latest source at https://downloads.mariadb.org and
the maria-discuss email list at https://launchpad.net/~maria-discuss

Please check all of the above before submitting a bug report
at http://mariadb.org/jira

root@hank2:/tmp#
```

I had to copy over the files manually because the complete package is too big. But it did run. Except that it apparently ran out of memory, which I think is just a configuration issue. So far so good.

Regards.
Seb

jow- commented:
Would you consider taking over maintainership of mysql/mariadb? In the past I merely took ownership of the package since it was a dependency of collectd.

lucize commented:
@micmac1 what is the final size of the smallest option ?
I added version 5.5 here https://github.com/openwrt/packages/pull/4221 because 10.x was too large and there were some bugs in disabling charsets and maria engine https://jira.mariadb.org/browse/MDEV-14078

Regards

micmac1 commented:
On Tue, Apr 03, 2018 at 08:40:19AM +0000, Jo-Philipp Wich wrote:
> Would you consider taking over maintainership of mysql/mariadb? In the
> past I merely took ownership of the package since it was a dependency
> of collectd.

Hello Jo,

Yes I would. Thank you for offering. I'll ping you when we have
something worth merging. I'll squash the commits before.

Kind regards,
Sebastian

micmac1 commented:
On Tue, Apr 03, 2018 at 09:54:08AM +0000, Lucian CRISTIAN wrote:
> @micmac1 what is the final size of the smallest option ?  I added
> version 5.5 here https://github.com/openwrt/packages/pull/4221 because
> 10.x was too large and there were some bugs in disabling charsets and
> maria engine https://jira.mariadb.org/browse/MDEV-14078

Hello Lucien,

As-is the server package is 9,0M on Mips, packed (ipk). I'm tinkering
with disabling Aria like you suggested in your upstream bug report. It
doesn't seem to be a problem (anymore) with 10.1.

But it doesn't seem to make much difference either size-wise:

-rwxr-xr-x 1 sk users 13502268  3. Apr 19:46 mariadb-10.1.32-orig/.pkgdir/mariadb-server/usr/bin/mysqld
-rwxr-xr-x 1 sk users 13498152  3. Apr 20:48 mariadb-10.1.32/.pkgdir/mariadb-server/usr/bin/mysqld

I'll try it out some more. I'll let you know. How big is 5.5?

Regards,
Seb

lucize commented:
extra charset takes a lot of space

ramips-mt7621
```
-rw-r--r-- 1 build build  463815 Sep  9 14:45 mariadb-client_5.5.57-1_mipsel_24kc.ipk
-rw-r--r-- 1 build build 4340486 Sep  9 14:45 mariadb-server_5.5.57-1_mipsel_24kc.ipk
```
x86_64
```
-rw-r--r-- 1 build build  186101 Sep  4 20:26 libmariadb_5.5.57-1_x86_64.ipk
-rw-r--r-- 1 build build  448553 Sep  4 20:26 mariadb-client_5.5.57-1_x86_64.ipk
-rw-r--r-- 1 build build  984969 Sep  4 20:26 mariadb-client-extra_5.5.57-1_x86_64.ipk
-rw-r--r-- 1 build build 4349105 Sep  4 20:26 mariadb-server_5.5.57-1_x86_64.ipk
```

micmac1 commented:
On Tue, Apr 03, 2018 at 07:14:39PM +0000, Lucian CRISTIAN wrote:
> extra charset takes a lot of space
> 
> ramips-mt7621
> ```
> -rw-r--r-- 1 build build  463815 Sep  9 14:45 mariadb-client_5.5.57-1_mipsel_24kc.ipk
> -rw-r--r-- 1 build build 4340486 Sep  9 14:45 mariadb-server_5.5.57-1_mipsel_24kc.ipk
> ```
> x86_64
> ```
> -rw-r--r-- 1 build build  186101 Sep  4 20:26 libmariadb_5.5.57-1_x86_64.ipk
> -rw-r--r-- 1 build build  448553 Sep  4 20:26 mariadb-client_5.5.57-1_x86_64.ipk
> -rw-r--r-- 1 build build  984969 Sep  4 20:26 mariadb-client-extra_5.5.57-1_x86_64.ipk
> -rw-r--r-- 1 build build 4349105 Sep  4 20:26 mariadb-server_5.5.57-1_x86_64.ipk
> ```

How about using EXTRA_CHARSETS=none, defaulting to UTF-8 and adding
extra packages for charsets that can be loaded dynamically? UTF-8 cannot
be loaded dynamically, so it's a good default (compiled-in).

Then the server package is down to a bit over 5 MByte:

sk@hotdamn ~/tmp/openwrt $ ls -l bin/packages/mips_24kc/packages/mariadb-server_10.1.32-1_mips_24kc.ipk
-rw-r--r-- 1 sk users 5381540  3. Apr 21:36 bin/packages/mips_24kc/packages/mariadb-server_10.1.32-1_mips_24kc.ipk

The dynamically loadable charsets are:

ls -lh usr/share/mysql/charsets/
insgesamt 232K
-rw-r--r-- 1 sk users  18K 26. Mär 16:18 Index.xml
-rw-r--r-- 1 sk users 1,8K 26. Mär 16:18 README
-rw-r--r-- 1 sk users 5,4K 26. Mär 16:18 armscii8.xml
-rw-r--r-- 1 sk users 5,4K 26. Mär 16:18 ascii.xml
-rw-r--r-- 1 sk users 8,1K 26. Mär 16:18 cp1250.xml
-rw-r--r-- 1 sk users 8,2K 26. Mär 16:18 cp1251.xml
-rw-r--r-- 1 sk users 5,5K 26. Mär 16:18 cp1256.xml
-rw-r--r-- 1 sk users 8,7K 26. Mär 16:18 cp1257.xml
-rw-r--r-- 1 sk users 5,4K 26. Mär 16:18 cp850.xml
-rw-r--r-- 1 sk users 5,4K 26. Mär 16:18 cp852.xml
-rw-r--r-- 1 sk users 5,5K 26. Mär 16:18 cp866.xml
-rw-r--r-- 1 sk users 6,4K 26. Mär 16:18 dec8.xml
-rw-r--r-- 1 sk users 5,4K 26. Mär 16:18 geostd8.xml
-rw-r--r-- 1 sk users 5,6K 26. Mär 16:18 greek.xml
-rw-r--r-- 1 sk users 5,4K 26. Mär 16:18 hebrew.xml
-rw-r--r-- 1 sk users 5,4K 26. Mär 16:18 hp8.xml
-rw-r--r-- 1 sk users 5,4K 26. Mär 16:18 keybcs2.xml
-rw-r--r-- 1 sk users 5,4K 26. Mär 16:18 koi8r.xml
-rw-r--r-- 1 sk users 6,4K 26. Mär 16:18 koi8u.xml
-rw-r--r-- 1 sk users 9,6K 26. Mär 16:18 latin1.xml
-rw-r--r-- 1 sk users 7,1K 26. Mär 16:18 latin2.xml
-rw-r--r-- 1 sk users 5,4K 26. Mär 16:18 latin5.xml
-rw-r--r-- 1 sk users 7,3K 26. Mär 16:18 latin7.xml
-rw-r--r-- 1 sk users 7,9K 26. Mär 16:18 macce.xml
-rw-r--r-- 1 sk users 7,9K 26. Mär 16:18 macroman.xml
-rw-r--r-- 1 sk users 6,4K 26. Mär 16:18 swe7.xml

Maybe it's not worth splitting them up. Maybe they should go into one
package. Maybe not.

If anybody says they want another charset included that is not loadable
dynamically, we could add it to EXTRA_CHARSETS or maybe just make it
available in menuselect as a non-default selection.

I saw that you split up the server package into smaller packages. I
think I'll copy that and do the same.

Best regards,
Seb

lucize commented:
this is my 10.2 try https://github.com/lucize/packages/commit/b1e2aea8f9bec716911d1b86436d1809f970fd28
regards

micmac1 commented:
Am April 3, 2018 7:50:48 PM UTC schrieb Lucian CRISTIAN <notifications@github.com>:
>this is my 10.2 try
>https://github.com/lucize/packages/commit/b1e2aea8f9bec716911d1b86436d1809f970fd28
>regards
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/openwrt/packages/pull/5851#issuecomment-378374236

I put up 10.2 at first, too. But Debian has the libatomic patches ready for MIPS and uses 10.1 in Stretch, so that was the lowest hanging fruit, so to speak.

micmac1 commented:
Am April 4, 2018 12:42:46 PM UTC schrieb patrikx3 <notifications@github.com>:
>Unfortunately, it is not building either ARM Cortexa 9 or MIPS (in
>Redis I replaced atomics to pthreads for mips, but my arm cortexa 9 is
>actually using atomics...)
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/openwrt/packages/pull/5851#issuecomment-378586268

Care to share the full log?

micmac1 commented:
Hello @lucize 

I've split up the binaries into utils, utils-extra, server and server-extra. I copied the distribution from the 10.2 Makefile you linked. Added some other changes from there as well (e.g. resolveip + m4).

Also added the package with the character sets.

Size-wise I get:

```
-rw-r--r-- 1 sk users 285K  4. Apr 22:57 bin/packages/mips_24kc/packages/libmariadbclient_10.1.32-1_mips_24kc.ipk
-rw-r--r-- 1 sk users 1,8M  4. Apr 22:57 bin/packages/mips_24kc/packages/mariadb-client-extra_10.1.32-1_mips_24kc.ipk
-rw-r--r-- 1 sk users 639K  4. Apr 22:57 bin/packages/mips_24kc/packages/mariadb-client_10.1.32-1_mips_24kc.ipk
-rw-r--r-- 1 sk users  24K  4. Apr 22:57 bin/packages/mips_24kc/packages/mariadb-extra-charsets_10.1.32-1_mips_24kc.ipk
-rw-r--r-- 1 sk users 7,5M  4. Apr 22:58 bin/packages/mips_24kc/packages/mariadb-server-extra_10.1.32-1_mips_24kc.ipk
-rw-r--r-- 1 sk users 5,3M  4. Apr 22:58 bin/packages/mips_24kc/packages/mariadb-server_10.1.32-1_mips_24kc.ipk
```

(and the small plugin packages)

I'd appreciate if you could give it a look, maybe even test it.

@p3x-robot is it possible the compile dies in the host pkg phase? Anyway, would be nice if you could provide the log.

Regards,
Seb

micmac1 commented:
It's for master. But ideally it shouldn't matter as the depends are available in 17.01. Please post the log if you still have it from your build attempt.

lucize commented:
@micmac1 so I tried a simple build

```
Package mariadb-client is missing dependencies for the following libraries:
libatomic.so.1
Makefile:449: recipe for target '/home/build/kirkwood/bin/packages/arm_xscale/packages/mariadb-client_10.1.32-1_arm_xscale.ipk' failed
make[2]: *** [/home/build/kirkwood/bin/packages/arm_xscale/packages/mariadb-client_10.1.32-1_arm_xscale.ipk] Error 1

```

micmac1 commented:
Am April 5, 2018 4:57:15 PM UTC schrieb Lucian CRISTIAN <notifications@github.com>:
>@micmac1 so I tried a simple build
>
>```
>Package mariadb-client is missing dependencies for the following
>libraries:
>libatomic.so.1
>Makefile:449: recipe for target
>'/home/build/kirkwood/bin/packages/arm_xscale/packages/mariadb-client_10.1.32-1_arm_xscale.ipk'
>failed
>make[2]: ***
>[/home/build/kirkwood/bin/packages/arm_xscale/packages/mariadb-client_10.1.32-1_arm_xscale.ipk]
>Error 1
>
>```
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/openwrt/packages/pull/5851#issuecomment-379004447

Interesting, thank you. I thought for arm there'd be no linking against libatomic.I'll check it out.

micmac1 commented:
Hello @lucize 

Should be fine now. Would be great if you could give it a test drive.

Hi @p3x-robot 
I ran an ARM Cortex 9 build and it went fine.

Kind regards,
Seb

lucize commented:
so I tried a minimal build but can't start it

```
root@ix2:/# mysql_install_db --force

FATAL ERROR: Could not find ./bin/my_print_defaults

```
otherwise seems to have been compiled ok

```
root@ix2:/# mysqld
2018-04-06 23:45:11 3070153904 [Note] Using unique option prefix 'key_buffer' is error-prone and can break in the future. Please use the full name 'key_buffer_size' instead.
2018-04-06 23:45:11 3070153904 [Note] Using unique option prefix 'key_buffer' is error-prone and can break in the future. Please use the full name 'key_buffer_size' instead.
2018-04-06 23:45:11 3070153904 [Note] mysqld (mysqld 10.1.32-MariaDB) starting as process 2274 ...
2018-04-06 23:45:11 3070153904 [Note] InnoDB: innodb_empty_free_list_algorithm has been changed to legacy because of small buffer pool size. In order to use backoff, increase buffer pool at least up to 20MB.

2018-04-06 23:45:11 3070153904 [Note] InnoDB: Using mutexes to ref count buffer pool pages
2018-04-06 23:45:11 3070153904 [Note] InnoDB: The InnoDB memory heap is disabled
2018-04-06 23:45:11 3070153904 [Note] InnoDB: Mutexes and rw_locks use GCC atomic builtins
2018-04-06 23:45:11 3070153904 [Note] InnoDB: GCC builtin __atomic_thread_fence() is used for memory barrier
2018-04-06 23:45:11 3070153904 [Note] InnoDB: Compressed tables use zlib 1.2.11
2018-04-06 23:45:11 3070153904 [Note] InnoDB: Using Linux native AIO
2018-04-06 23:45:11 3070153904 [Note] InnoDB: Using generic crc32 instructions
2018-04-06 23:45:11 b6fed4b0 InnoDB: Error: Linux Native AIO interface is not supported on this platform. Please check your OS documentation and install appropriate binary of InnoDB.
InnoDB: You can disable Linux Native AIO by setting innodb_use_native_aio = 0 in my.cnf
2018-04-06 23:45:11 3070153904 [Warning] InnoDB: Linux Native AIO disabled.
2018-04-06 23:45:11 3070153904 [Note] InnoDB: Initializing buffer pool, size = 128.0M
2018-04-06 23:45:11 3070153904 [Note] InnoDB: Completed initialization of buffer pool
2018-04-06 23:45:11 3070153904 [Note] InnoDB: The first specified data file ./ibdata1 did not exist: a new database to be created!
2018-04-06 23:45:11 3070153904 [Note] InnoDB: Setting file ./ibdata1 size to 12 MB
2018-04-06 23:45:11 3070153904 [Note] InnoDB: Setting log file ./ib_logfile101 size to 48 MB
2018-04-06 23:45:11 3070153904 [Note] InnoDB: Setting log file ./ib_logfile1 size to 48 MB
2018-04-06 23:45:12 3070153904 [Note] InnoDB: Renaming log file ./ib_logfile101 to ./ib_logfile0
2018-04-06 23:45:12 3070153904 [Warning] InnoDB: New log files created, LSN=45781
2018-04-06 23:45:12 3070153904 [Note] InnoDB: Doublewrite buffer not found: creating new
2018-04-06 23:45:12 3070153904 [Note] InnoDB: Doublewrite buffer created
2018-04-06 23:45:12 3070153904 [Note] InnoDB: 128 rollback segment(s) are active.
2018-04-06 23:45:12 3070153904 [Warning] InnoDB: Creating foreign key constraint system tables.
2018-04-06 23:45:12 3070153904 [Note] InnoDB: Foreign key constraint system tables created
2018-04-06 23:45:12 3070153904 [Note] InnoDB: Creating tablespace and datafile system tables.
2018-04-06 23:45:12 3070153904 [Note] InnoDB: Tablespace and datafile system tables created.
2018-04-06 23:45:12 3070153904 [Note] InnoDB: Waiting for purge to start
2018-04-06 23:45:12 3070153904 [Note] InnoDB:  Percona XtraDB (http://www.percona.com) 5.6.38-83.0 started; log sequence number 0
2018-04-06 23:45:12 3070153904 [ERROR] Could not open mysql.plugin table. Some plugins may be not loaded
2018-04-06 23:45:12 3070153904 [ERROR] Can't open and lock privilege tables: Table 'mysql.servers' doesn't exist
2018-04-06 23:45:12 3070153904 [Note] Server socket created on IP: '127.0.0.1'.
2018-04-06 23:45:12 2741161256 [Note] InnoDB: Dumping buffer pool(s) not yet started
2018-04-06 23:45:12 3070153904 [ERROR] Fatal error: Can't open and lock privilege tables: Table 'mysql.user' doesn't exist

```
regards

micmac1 commented:
I ran into that, too. Needed to specify the basedir on the command line. I think you can also put it into my.cnf.

lucize commented on utils/mariadb/Makefile:
```diff
@@ -0,0 +1,539 @@
+#
+# Copyright (C) 2018 OpenWrt.org
```
Usually you put your name here

lucize commented on utils/mariadb/Makefile:
```diff
@@ -0,0 +1,539 @@
+MARIADB_SERVER_EXTRA := \
+	aria* \
+	msql2mysql \
+	my_print_defaults \
```
can you move this to basic so mysql_install_db --force won't give fatal error ?

lucize replied:
I was thinking only for my_print_defaults

micmac1 replied:
Got you, thanks!

micmac1 commented:
Hello @p3x-robot 

Thanks! Word of warning, I removed "+libpthread" from the DEPENDS as the dep is not needed anymore in OpenWrt master (OpenWrt adds the dep automatically). But I'm not sure LEDE 17.01 does the same. So if you see an error there due do a missing depend on libpthread, just add +libpthread to MARIADB_COMMON_DEPENDS.

Kind regards,
Seb

micmac1 commented:
On Sat, Apr 07, 2018 at 10:02:51AM +0000, patrikx3 wrote:
> Saying:
> ```Makefile:35: *** recipe commences before first target.  Stop.```

You need to add a backslash after +libpthread :) Check the other lines.

lucize commented:
@micmac1 I think there is a problem with the `mysql_install_db `script, it still can't find the `my_print_defaults` or maybe the my.cnf doesn't have all the necessary options, I'll look later on my PR for the my.cnf and the startup

micmac1 commented:
Good morning @lucize and @p3x-robot 

I plan on asking that this gets merged tonight, because I would like this to be included in the coming release. I'll do a test compile until then for PPC (I did that for ARC, ARM, MIPS and x86 so far). The way I see it the binaries are working OK. The default config and init script could maybe use some brushing up, but that shouldn't stop us from getting the packages into the tree now.

Anyway, thanks again for your help, I really appreciate it! And if you have anything you want changed just ping me.

Best regards,
Seb

micmac1 commented:
Hello @jow- 

Please merge this into the tree. What I initially mentioned regarding python-mysql not compiling is not a problem anymore. The mariadb version was downgraded from the initial 10.2 to 10.1 (current Debian stable mariadb) and python-mysql compiles fine against it.

With kind regards,
Seb

jow- commented:
We need to factor out the underscores in the plugin package names; underscores should not be used in package bas names as they're used as semantic separators by opkg.

micmac1 commented:
Hello Jo,

Thanks for letting me know. I'll take care of it when I get home from work.

Kind regards,
Seb

micmac1 commented:
Hello @jow- 

The plugin naming issue has been resolved.

Kind regards,
Seb

micmac1 commented:
Hello @jow- 

I don't mean to be pushy, but it would be good if this could go in before a new branch for the new release is cut. :-) Just my 2 cents.

Kind regards,
Seb

micmac1 commented:
ping @jow- :-)

lucize commented:
@p3x-robot you can see that in the last year or so only updates are accepted and very few exceptions of new packages, so have patience, maybe they have a reason !

micmac1 commented:
Am April 23, 2018 6:34:57 PM UTC schrieb patrikx3 <notifications@github.com>:
>@lucize no problem, i have built for openwrt, but i was hoping to not
>have to work on it, but openwrt is awesome, i can just add a feed and
>it works, ok!!!
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/openwrt/packages/pull/5851#issuecomment-383677832

:-) Thanks for the support, but I'm sure that Jo is just busy with preparing the next OpenWrt release.

jow- commented:
Merged, thanks @micmac1 !
